src/App.tsx

```diff
@@ -14,11 +14,98 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
 
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  selectHandler = (item: string) => {
+    const { selectedGoods } = this.state;
+    const copy = [...selectedGoods, item];
+
+    this.setState({ selectedGoods: copy });
+  };
```

Comment:
also, it would be nice to use functional setState
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

```suggestion
  handleSelect = (item: string) => {
    this.setState((prevState) => ({ selectedGoods: [...prevState.selectedGoods, item] }));
  };
```